### PR TITLE
Upgrade dart sys, upgrade ci version, and fix linters

### PR DESCRIPTION
**Pull request by rogurotus**
Dart-sys update

Should fix the assembly of MacOs and android from under MacOs. #1021 #1022

also string refactoring

## Checklist

- [x] An issue to be fixed by this PR is listed above.
- [x] New tests are added to ensure new features are working. End-to-end tests are usually in the `./frb_example/pure_dart` example, more specifically, `rust/src/api.rs` and `dart/lib/main.dart`.
- [x] The code generator is run and the code is formatted (e.g. via `just refresh_all`).
- [x] If this PR adds/changes features, documentations (in the `./book` folder) are updated.
- [x] CI is passing.

### CODE REVIEW

**Comment by rogurotus** [comment]
In principle, it works, but there is a strange behavior on MacOs flutter 3.7.0, but it seems to be not related to dart-sys.

**Comment by fzyzcjy** [comment]
> there is a strange behavior on MacOs flutter 3.7.0, but it seems to be not related to dart-sys.

Curious what is that?

**Comment by rogurotus** [comment]
the integration test just hangs and some internal timeout crashes.
I would explore more but I don't have a Mac

```
loading /Users/runner/work/flutter_rust_bridge/flutter_rust_bridge/frb_example/with_flutter/integration_test/main.dart (failed)
  TimeoutException after 0:12:00.000000: Test timed out after 12 minutes.
  package:test_api/src/backend/invoker.dart 334:28  Invoker._handleError.<fn>
  dart:async/zone.dart 1390:47                      _rootRun
  dart:async/zone.dart 1300:19                      _CustomZone.run
  package:test_api/src/backend/invoker.dart 332:10  Invoker._handleError
  package:test_api/src/backend/invoker.dart 288:9   Invoker.heartbeat.<fn>.<fn>
  dart:async/zone.dart 1398:13                      _rootRun
  dart:async/zone.dart 1300:19                      _CustomZone.run
  package:test_api/src/backend/invoker.dart 287:38  Invoker.heartbeat.<fn>
  dart:async-patch/timer_patch.dart 18:15           Timer._createTimer.<fn>
  dart:isolate-patch/timer_impl.dart 398:19         _Timer._runTimers
  dart:isolate-patch/timer_impl.dart 429:5          _Timer._handleMessage
  dart:isolate-patch/isolate_patch.dart 192:26      _RawReceivePort._handleMessage
  ```

**Comment by fzyzcjy** [comment]
That's weird. I will examine it later

File: .github/workflows/flutter_macos_test.yml [hunk 1]
```diff
@@ -4,7 +4,7 @@ on:
       flutter_version:
         required: false
         type: string
-        default: '3.3.1'
+        default: '3.3.10'
```
Comment (fzyzcjy) [inline .github/workflows/flutter_macos_test.yml:7]:
```suggestion
        # TODO should be 3.7.0 but has a weird bug currently https://github.com/fzyzcjy/flutter_rust_bridge/pull/1023#issuecomment-1415852845
        default: '3.3.10'
```

**Review by fzyzcjy — APPROVED** [review]

**Comment by fzyzcjy** [comment]
Good job!

**Pull request merged**